pilot/cmd/pilot-agent/app/wait.go

```diff
@@ -41,17 +42,23 @@ var (
 			log.Infof("Waiting for Envoy proxy to be ready (timeout: %d seconds)...", timeoutSeconds)
 
 			var err error
-			timeoutAt := time.Now().Add(time.Duration(timeoutSeconds) * time.Second)
-			for time.Now().Before(timeoutAt) {
-				err = checkIfReady(client, url)
-				if err == nil {
-					log.Infof("Envoy is ready!")
-					return nil
+			go func() {
+				time.Sleep(time.Duration(timeoutSeconds) * time.Second)
```

Comment:
No need start a thread.

ch := time.After()

Reply:
okay, I will remove the unnecessary thread and use `time.After`, thanks @hzxuzhonghu!